Core/Src/main.c

```diff
@@ -87,14 +90,49 @@ int main(void)
   /* Initialize all configured peripherals */
   MX_GPIO_Init();
   MX_USART2_UART_Init();
+  MX_SPI1_Init();
+  MX_TIM1_Init();
   /* USER CODE BEGIN 2 */
+  /* Pull CS line to default high to ADC (pin 8) */
+  HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, GPIO_PIN_SET);
+
+  /* Start timer */
+  HAL_TIM_PWM_Start(&htim1, TIM_CHANNEL_1);
```

Comment:
make sure to set the value in the ccr such that the timer generates a valid pwm signal before calling HAL_TIM_PWM_Start